Feature/single folder build

beesaycheese opened the pull request:
Mac:
```
git clone https://github.com/datature/portal
chmod u+x ./setup-virtualenv.sh && ./setup-virtualenv.sh
. portal-build/.venv/bin/activate
python3 portal-build/portal.py (additional flags include --electron and --gpu)
```
Win:
```
git clone https://github.com/datature/portal
setup-virtualenv.sh
portal-build\.venv\Scripts\activate
python portal-build\portal.py  (additional flags include --electron and --gpu)
```

Additional Stuff changed:
1. Toaster logic (On browser mode, the blue toaster will stay forever until it detects the backend)
2. Set a global isConnected state
3. Disable inference buttons when there is no loaded model. On hover, a tooltip info will appear
4. OnComponentDidMount, the model component will call the get model list api until it gets a postive ans. During the wait, the model button is in skeleton mode
5. GPU feature to request for a restart

beesaycheese committed:
Change location of next export to portal-build/out

beesaycheese committed:
Update piplock file

beesaycheese committed:
Adjust main script to accomodate for portalbuild

beesaycheese committed:
Create a setup-virtuelenv bash script

beesaycheese committed:
add portal-build to gitignore

beesaycheese committed:
adjust root package.json for running electron in portal-build

beesaycheese committed:
portal.py and setup.sh

beesaycheese committed:
Remove unnecessary imports

beesaycheese committed:
Update setup python command for windows

beesaycheese committed:
Adjust runtime to consider non-electron cases

beesaycheese committed:
Adjust the portal.py to run with the one port only

beesaycheese committed:
Shift portal.py to the root folder

beesaycheese committed:
Add portal.py to portal-build for global setup

beesaycheese committed:
Generalise toast to add additional feature

beesaycheese committed:
Adjust runtime toast and set a global isConnected state

beesaycheese committed:
Fix logic error in the disable button

beesaycheese committed:
Add ToolTip hint for inference

beesaycheese committed:
Add styling for the tooltip icon

beesaycheese committed:
Remove hint icon and replace it with a hover

beesaycheese committed:
Retain the blue toaster forever on the browser mode

beesaycheese committed:
Fix backend error

beesaycheese committed:
Set model's skeleton state

beesaycheese committed:
Adjust the size of the spinner icon

virtuoso98 commented on src/app/pages/index.js:
```diff
@@ -24,8 +24,9 @@ class Annotation extends React.Component {
     /* Global Setting / Styling State */
     this.state = {
-      useDarkTheme: false,
+      useDarkTheme: true,
```
Is the default now dark theme?

beesaycheese replied:
Yeap

virtuoso98 commented on src/app/src/components/annotations/menu.jsx:
```diff
@@ -360,29 +364,40 @@ export default class AnnotationMenu extends Component {
+                      ? "bp3-active"
+                      : ""
+                  }
+                  onClick={() => this.props.callbacks.GetInference()}
+                />
+                <MenuItem
+                  disabled={!this.props.isConnected || !this.props.loadedModel}
+                  icon="heat-grid"
+                  text="Bulk Analysis"
+                  label={<KeyCombo combo="B" />}
+                  className={
+                    this.props.userEditState === "Bulk Analysis"
+                      ? "bp3-active"
+                      : ""
+                  }
+                  onClick={() => this.props.callbacks.GetInference()}
```
Why is bulk analysis and single inference pointing towards the same function? 

Note: ignore this if this is addressed in another PR.

beesaycheese replied:
Yeap, it's in another PR that's not merged yet

virtuoso98 replied:
Got it!

virtuoso98 commented:
approving this PR

beesaycheese committed:
Remove unused import

marcus-neo committed:
gpu switch support

added routes to set/clear/get gpu flag
based on gpu flag, alter the cuda_visible environment variable.

marcus-neo committed:
accommodate front-end crashes but backend didn't crash scenario

beesaycheese committed:
Merge pull request #22 from datature/origin/feature/gpu-switch

Backend GPU implementation

beesaycheese committed:
Change the route of the use_gpu

beesaycheese committed:
DEfault to gpu if there is no use_gpu file

beesaycheese committed:
Add new api call

beesaycheese committed:
Handle gpu on the frontend for electron and browser

beesaycheese committed:
Fix logic error

beesaycheese committed:
Fix gpu to accommodate for portal-build

beesaycheese committed:
Fix relative pathing issues

beesaycheese committed:
Conditional delete cache in shutdown route

beesaycheese committed:
Merge branch 'develop' of https://github.com/datature/portal into feature/single-folder-build

beesaycheese committed:
Bugfix for unnecesssary gpu calls

beesaycheese committed:
Merge branch 'develop' of https://github.com/datature/portal into feature/single-folder-build

beesaycheese committed:
Fix DArk mode styling

beesaycheese committed:
remove progress toast for selectAsset

beesaycheese committed:
Fix bulk analysis bug
